.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,29 @@
+name: codeql
+
+# FIXME uncomment conditions before merging PR
+on:
+  push:
+    # branches:
+    # - master
+  pull_request:
+    # branches:
+    # - master
+  # schedule:
+  #   - cron: "0 3 * * 4"
+
+jobs:
+  analyze:
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: go
+    - name: Perform CodeQL Analysis
```

Comment:
do we need to add step to do either autobuild or manual build before this step ?

Reply:
I don't think it's necessary. CodeQL does source code analysis only, it doesn't use any build artifacts. IIUC All the generated source code is already checked into Cilium's repo so no build steps are required.